bundles/org.openhab.binding.ecovacs/src/main/java/org/openhab/binding/ecovacs/internal/EcovacsBindingConstants.java

```diff
@@ -0,0 +1,158 @@
+    public static final String CHANNEL_ID_SUCTION_POWER = "settings#suction-power";
+    public static final String CHANNEL_ID_TOTAL_CLEANING_TIME = "total-stats#total-cleaning-time";
+    public static final String CHANNEL_ID_TOTAL_CLEANED_AREA = "total-stats#total-cleaned-area";
+    public static final String CHANNEL_ID_TOTAL_CLEAN_RUNS = "total-stats#total-clean-runs";
+    public static final String CHANNEL_ID_TRUE_DETECT_3D = "settings#true-detect-3d";
+    public static final String CHANNEL_ID_VOICE_VOLUME = "settings#voice-volume";
+    public static final String CHANNEL_ID_WATER_PLATE_PRESENT = "status#water-system-present";
+    public static final String CHANNEL_ID_WATER_AMOUNT = "settings#water-amount";
+    public static final String CHANNEL_ID_WIFI_RSSI = "status#wifi-rssi";
+
+    public static final String CHANNEL_TYPE_ID_CLEAN_MODE = "current-cleaning-mode";
+    public static final String CHANNEL_TYPE_ID_LAST_CLEAN_MODE = "last-clean-mode";
+
+    public static final String CMD_AUTO_CLEAN = "clean";
+    public static final String CMD_PAUSE = "pause";
+    public static final String CMD_RESUME = "resume";
+    public static final String CMD_CHARGE = "charge";
+    public static final String CMD_STOP = "stop";
+    public static final String CMD_SPOT_AREA = "spotArea";
+    public static final String CMD_CUSTOM_AREA = "customArea";
+
+    public static class StateOptionEntry<T extends Enum<T>> {
+        public final T enumValue;
+        public final String value;
+        public final Optional<DeviceCapability> capability;
+
+        StateOptionEntry(T enumValue, String value) {
+            this(enumValue, value, null);
+        }
+
+        StateOptionEntry(T enumValue, String value, @Nullable DeviceCapability capability) {
+            this.enumValue = enumValue;
+            this.value = value;
+            this.capability = Optional.ofNullable(capability);
+        }
+    }
+
+    public static class StateOptionMapping<T extends Enum<T>> extends HashMap<T, StateOptionEntry<T>> {
+        private static final long serialVersionUID = -6828690091106259902L;
+
+        public String getMappedValue(T key) {
+            StateOptionEntry<T> entry = get(key);
+            if (entry != null) {
+                return entry.value;
+            }
+            throw new IllegalArgumentException("No mapping for key " + key);
+        }
+
+        public @Nullable T findMappedEnumValue(String value) {
+            return entrySet().stream().filter(entry -> entry.getValue().value.equals(value))
+                    .map(entry -> entry.getKey()).findFirst().orElse(null);
+        }
+
+        @SafeVarargs
+        public static <T extends Enum<T>> StateOptionMapping<T> of(StateOptionEntry<T>... entries) {
+            StateOptionMapping<T> map = new StateOptionMapping<>();
+            for (StateOptionEntry<T> entry : entries) {
+                map.put(entry.enumValue, entry);
+            }
+            return map;
+        }
+    }
```

Comment:
Consider moving to another file to not clutter this class intended for constants.